SeafarersAPI/Data/Repositories/SubmissionRepository.cs

```diff
@@ -3,56 +3,57 @@
 using System;
 using System.Collections.Generic;
 using System.Linq;
-using System.Threading.Tasks;
 
 namespace SeafarersAPI.Data.Repositories
 {
-    public class SubmissionRepository
+    public class SubmissionRepository : IRepository<Submission>
     {
         public SubmissionContext context;
         public SubmissionRepository()
         {
-            if(context== null)
+            if(context == null)
             {
                 context = new SubmissionContext();
             }
         }
 
-        public List<Submission> GetSubmissions()
+        public IEnumerable<Submission> GetList(int startIndex, int amountToGet)
         {
-            return context.Submissions.ToList();
+            return context.Submissions.Skip(startIndex).Take(amountToGet).ToList();
         }
 
-        public Submission GetSubmissionById(int submissionId)
+        public int Save(Submission submission)
         {
-            var submission = context.Submissions.Where(x => x.SubmissionId == submissionId).SingleOrDefault();
-
-            return submission;
+            // TODO: Add locking mechanism or we'll run into concurrency issues
+            context.Submissions.Add(submission);
+            return context.SaveChanges();
```

Comment:
We want to return the results of the save here so that we can handle errors in the case nothing was saved. We also want to add a locking mechanism to avoid concurrency problems in the future.

Reply:
Does this return the submission id so that we can display it to the user on the confirmation page?

Reply:
No it's the count of the number of updated items in the Database. We could however modify the controller to send us the Id's or something similar. That wouldn't be the repositories job.

Reply:
Can we add that to the controller now? I think it can simply return the SubmissionID for now

Reply:
I agree. This API was written in a rush. So, there is a lot improve.